src/tools/compiletest/src/main.rs

```diff
@@ -524,3 +526,24 @@ fn extract_lldb_version(full_version_line: Option<String>) -> Option<String> {
     }
     None
 }
+
+fn detect_node() -> Option<String> {
+    // Look for `nodejs` or `node` on PATH in that order, returning the name
+    // of whichever is found.
+    let path = match env::var_os("PATH") {
+        Some(x) => x,
+        None => return None
+    };
+
+    for mut dir in env::split_paths(&path) {
+        for bin in &["nodejs", "node"] {
+            dir.push(bin);
+            if dir.exists() {
+                return Some(bin.to_string());
+            }
+            dir.pop();
+        }
+    }
+    None
+}
```

Comment:
Yeah, I forgot about Windows so at least the .exe variants should be added. I also considered trying to run each instead of scanning PATH, which would free us from needing to know about a platform's executable suffix so I think that's a better solution.